### DISCUSSION TITLE
Different paths used to store files on paste image from clipboard

### DISCUSSION THREAD

**Discussion by doctomlug** [freeplane/freeplane#2075] [Help] [answered]
Hello,

I am using FreePlane (1.12.6) on different computers running win10 or win11. Recently the win11 machine started to behave differently when images are pasted into a saved map.

"usual" behavior (still working on the win10 machine):
Image gets pasted into a child node, a file dialog asks for the filename, file gets save under subdir "./[FILENAME]_files"

"new" behavior
Image gets pasted into a child node, NO dialog pops up (generally a good thing), image gets saved in some strange directory
$USER\Appdata\Local\Packages\MicrosoftWindows.Client.CBS_cw5n1h2txyewy\TempState\ScreenClip

The "usual" behavior is the wanted one as I copy maps regularly between computers / users and lose the images if they are not saved in the subdir next to the mm-file. Furthermore all the images in my "old" maps are shown as error message only (expected as the path is not matching any more)

Before this switch in behavior happened, I changed some of the settings and played around with the developer addon. Trying to revert the changes or even deleting all settings to get back to a default fresh start (feature mm opens as first map) did not change it back to the "usual" one. Image caching setting "Disk" or "Ram" has no effect. I compared saved preferences from computers that show different behavior and found no major difference.

I am lost and out of ideas: What to do to get the "usual" behavior back?

**Comment by i-plasm** [comment, marked as answer]
It seems that the strange path is designated by windows Snip and sketch tool. If that's your case, then it does not have to do with Freeplane. I suspect that something about the tool settings on your Windows 11 and/or something about your copy-paste workflow is causing the issue. Try adjusting your tool options - I don't use that tool, but you can look for options related to autosave or clipboard - and/or your copy-paste workflow to make sure an actual image gets copied to clipboard (not a file).

**Reply by doctomlug** [reply]
That was a hint in the right direction, culprit seams to be the win snipping tool. All documented ways to change the tool's settings did not work on my device (company computer). The solution that worked in the end was installing greenshot to capture screen areas.

I found no way to figure out if the clipboard contains the file or the picture but assume it's the file based on freeplane's behavior. Greenshot allows to configure this and it works as "usual".

Is there a way to pasting images without the file dialog and just use the [FILENAME]_file subdir and the numerical image filename automatically?

**Reply by grotesque** [reply]
Thanks for giving a solution to this.
@dpolivaev can we easily paste clipboard images into freeplane without the file dialog and instead the files being saved to default [mmfilename]_files folder located in the same dir as the mm file?
